src/charm.py

```diff
@@ -316,5 +338,41 @@ def _on_remove(self, _):
         self.unit.status = MaintenanceStatus("K8s resources removed")
 
 
+def sort_sidebar_items(sidebar_items: List[SidebarItem], preferred_links: List[str]):
+    """Sorts a list of SidebarItems by their link text, moving preferred links to the top.
+
+    The sorted order of the returned list will be:
+    * any links who have a text field that matches a string in `preferred_link_text`, in the order
+      specified in preferred_link_text
+    * any remaining links, in alphabetical order
+
+    For example, if:
+      sidebar_items=[SidebarItem(text="1"...), SidebarItem(text="2"...), SidebarItem(text="3"...)]
+      preferred_link_text=["2", "4"]
+    The return will be:
+      sidebar_items=[SidebarItem(text="2"...), SidebarItem(text="1"...), SidebarItem(text="3"...)]
+
+    Args:
+        sidebar_items: List of SidebarItems to sort
+        preferred_links: List of strings of SidebarItem text values
+
+    Returns:
+        Ordered list of SidebarItems
+    """
+    sidebar_items_lookup = {item.text: item for item in sidebar_items}
```

Comment:
I just wonder cant we have two items with same name ? ... this will have problems in that case. Should it report some error ?

Reply:
Yeah good call.  I was thinking to ignore it in the code (if the user defines two links with the same name, they've done it for a reason) - but the implementation details here break that!  I'll fix